src/model/bot.py

```diff
@@ -514,6 +562,147 @@ def toggle_auto_retaliate(self, toggle_on: bool):
         else:
             self.log_msg("Auto retaliate is already off.")
 
+    def toggle_melee_attack(self, toggle_on: bool):
```

Comment:
A good rule of thumb in code is that when there is duplicated code, it can probably be compressed into a single function. I think that can be done here. Also, I don't think it should be a toggle, since the actual attack styles can't be toggled off - the only way to turn a style off is to select a different one. Also I think `controlled` should be `shared`, if we are going for xp-type rather than attack type (which would be `accurate`, `aggressive`, `defensive`, `controlled`.)

Maybe the function can be something like:

```python
def select_melee_style(self, xp_type: str)
  """
  Args:
    xp_type: the attack type ("attack", "strength", "defence", or "shared").
  """
  # Click the combat tab
  ...
  # Define a list of all possible weapons
  weapons = ['axe', 'pickaxe', 'bulwark', 'claw', 'dagger', 'mace', 'maul', 'whip', 'banner', 'hally', 'spear', 'long_sword']
  
  for weapon in weapons:
    if result := imsearch.search_img_in_rect(imsearch.BOT_IMAGES.joinpath("melee_combat", "attack", f"{weapon}_off.png"), self.win.control_panel, 0.05):
      self.mouse.move_to(result.random_point(), mouseSpeed='medium')
      self.mouse.click()
      self.log_msg(f"Attack style '{xp_type}' selected."
      return
  self.log_msg(f"Attack style '{xp_type}' is already selected.")
```

This way, you don't need the `off` folder. You also wouldn't really need to have `_off` in the name of each png.

Thoughts?